server/graphql/document/src/types/document.rs

```diff
@@ -0,0 +1,112 @@
+use async_graphql::*;
+use async_graphql::{dataloader::DataLoader, Context};
+use chrono::{DateTime, Utc};
+
+use graphql_core::{standard_graphql_error::StandardGraphqlError, ContextExt};
+use repository::Document;
+use service::document::raw_document::RawDocument;
+
+use crate::json_schema_loader::JsonSchemaLoader;
+
+use super::json_schema::JSONSchemaNode;
+
+pub struct DocumentNode {
+    pub document: Document,
+}
+
+#[Object]
+impl DocumentNode {
+    pub async fn id(&self) -> &str {
+        &self.document.id
+    }
+
+    pub async fn name(&self) -> &str {
+        &self.document.name
+    }
+
+    pub async fn parents(&self) -> &[String] {
+        &self.document.parents
+    }
+
+    pub async fn author(&self) -> &str {
+        &self.document.author
+    }
+
+    pub async fn timestamp(&self) -> &DateTime<Utc> {
+        &self.document.timestamp
+    }
+
+    pub async fn r#type(&self) -> &str {
+        &self.document.r#type
+    }
+
+    pub async fn data(&self) -> &serde_json::Value {
+        &self.document.data
+    }
+
+    pub async fn schema(&self, ctx: &Context<'_>) -> Result<Option<JSONSchemaNode>> {
+        Ok(match &self.document.schema_id {
+            Some(schema_id) => {
+                let loader = DataLoader::new(
```

Comment:
Why is data loader instantiated here, i though it had to be shared instance to use data loader functionality